packages/session-replay-browser/src/config/remote-config.ts

```diff
@@ -0,0 +1,121 @@
+const SUCCESS_REMOTE_CONFIG = 'Session replay remote config successfully fetched';
+const MAX_RETRIES_EXCEEDED_MESSAGE = 'Session replay remote config fetch rejected due to exceeded retry count';
+export const REMOTE_CONFIG_SERVER_URL = 'https://sr-config.amplitude.com/config';
+export const REMOTE_CONFIG_SERVER_URL_STAGING = 'https://sr-client-cfg.stag2.amplitude.com/config';
+
+export class SessionReplayRemoteConfigFetch implements ISessionReplayRemoteConfigFetch {
+  localConfig: ISessionReplayLocalConfig;
+  sessionIDBStore: AmplitudeSessionReplaySessionIDBStore;
+  retryTimeout = 1000;
+  attempts = 0;
+  lastFetchedSessionId: number | undefined;
+  sessionTargetingMatch = false;
+
+  constructor({
+    localConfig,
+    sessionIDBStore,
+  }: {
+    localConfig: ISessionReplayLocalConfig;
+    sessionIDBStore: AmplitudeSessionReplaySessionIDBStore;
+  }) {
+    this.localConfig = localConfig;
+    this.sessionIDBStore = sessionIDBStore;
+  }
+
+  getRemoteConfig = async (sessionId: number): Promise<ISessionReplayRemoteConfig | void> => {
+    // Then check IndexedDB for session
+    const remoteConfig = await this.sessionIDBStore.getRemoteConfigForSession(sessionId);
+    if (remoteConfig) {
+      return remoteConfig;
+    }
+    // Finally fetch via API
+    return this.fetchRemoteConfig(sessionId);
+  };
+
+  getSamplingConfig = async (sessionId: number): Promise<SamplingConfig | void> => {
+    const remoteConfig = await this.getRemoteConfig(sessionId);
+    return remoteConfig?.sr_sampling_config;
+  };
+
+  getServerUrl() {
+    if (this.localConfig.serverZone === ServerZone.STAGING) {
+      return REMOTE_CONFIG_SERVER_URL_STAGING;
+    }
+
+    return REMOTE_CONFIG_SERVER_URL;
+  }
+
+  fetchRemoteConfig = async (sessionId: number): Promise<ISessionReplayRemoteConfig | void> => {
+    if (sessionId === this.lastFetchedSessionId && this.attempts >= this.localConfig.flushMaxRetries) {
+      return this.completeRequest({ err: MAX_RETRIES_EXCEEDED_MESSAGE });
+    } else if (sessionId !== this.lastFetchedSessionId) {
+      this.lastFetchedSessionId = sessionId;
+      this.attempts = 0;
+    }
+
+    try {
+      const urlParams = new URLSearchParams({
+        api_key: this.localConfig.apiKey,
+        config_keys: 'sessionReplay',
+      });
+      const options: RequestInit = {
+        headers: {
+          'Content-Type': 'application/json',
+          Accept: '*/*',
+        },
+        method: 'GET',
+      };
+      const server_url = `${this.getServerUrl()}?${urlParams.toString()}`;
+      this.attempts += 1;
+      const res = await fetch(server_url, options);
+      if (res === null) {
+        return this.completeRequest({ err: UNEXPECTED_ERROR_MESSAGE });
+      }
+      const parsedStatus = new BaseTransport().buildStatus(res.status);
+      switch (parsedStatus) {
+        case Status.Success:
+          return this.parseAndStoreConfig(sessionId, res);
+        case Status.Failed:
```

Comment:
Should we also retry on 429 or is that already built in?

Reply:
We made a decision with @jackson-amplitude a while ago to specifically only retry on 500, this was from a time of more uncertainty on our system stability, but I think it's ok to stick with for now. We could loosen up a bit if we decide we are more certain in our stability in upcoming weeks